src/lib.rs

```diff
@@ -770,3 +770,9 @@ impl Module {
         }
     }
 }
+
+fn limited_string(len: usize, u: &mut Unstructured) -> Result<String> {
+    let mut ret: String = u.arbitrary()?;
+    ret.truncate(len);
+    Ok(ret)
```

Comment:
Let's do what `Arbitrary for String` does but with a max size:

```suggestion
    const MAX_SIZE: usize = 1_000;

    let size = u.arbitrary_len::<u8>()?;
    let size = std::cmp::min(size, MAX_SIZE);
    match str::from_utf8(&u.peek_bytes(size).unwrap()) {
        Ok(s) => {
            u.get_bytes(size).unwrap();
            Ok(s.into())
        }
        Err(e) => {
            let i = e.valid_up_to();
            let valid = u.get_bytes(i).unwrap();
            let s = unsafe {
                debug_assert!(str::from_utf8(valid).is_ok());
                str::from_utf8_unchecked(valid)
            };
            Ok(s.into())
        }
    }
```

Reply:
This will avoid consuming all the raw input that would have been part of the string if we hadn't been imposing a limit, so that it can be used by later calls to `Unstructured` methods.

Reply:
Updated!